### PR TITLE
fix: Interfaceless providers

### PR DESCRIPTION
**Pull request by jlacivita**
Allows an API to provide a capability with no provider interface, e.g. for push/pull APIs.

### CODE REVIEW

File: util/shared/modules.mjs [hunk 1]
```diff
@@ -54,26 +54,33 @@ const getMethods = compose(
...
 
-const isProviderMethod = compose(
-    option(false),
-    map(_ => true),
-    chain(
-      find(
-        and(
-          propEq('name', 'capabilities'),
-          propSatisfies('x-provides', not(isEmpty))
+const isProviderInterfaceMethod = compose(
```
Comment (cletustboone) [inline util/shared/modules.mjs:57]: 💪🏻

**Comment by cletustboone** [comment]
I'll ask the dumb question. Why are push/pull APIs interfaceless?

**Review by cletustboone — APPROVED** [review]

File: src/js/shared/Events/index.mjs [hunk 2]
```diff
@@ -229,7 +229,6 @@ export const prioritize = function(...args) {
 }
 
 const unsubscribe = (key) => {
-  console.log(key)
```
Comment (cletustboone) [inline src/js/shared/Events/index.mjs:232]: Do you really want this `console.log` here?

**Comment by jlacivita** [comment]
Techincally, they're "Provider-interfaceless" not "interfaceless"

A provider interface is an interface that the app implements with their own custom code, so we can call APIs on it. Effectively a reverse interface, made up of a bunch of private events that trigger calling the APIs, and provide response/error methods for each Provider Interface API.

A push/pull API is just a normal interface that is part of the RPC interface described by our OpenRPC docs.

File: src/js/shared/Events/index.mjs (hunk 2 above)
Comment (jlacivita) [reply src/js/shared/Events/index.mjs:232]: no, that's why I took it out.

Comment (cletustboone) [reply src/js/shared/Events/index.mjs:232]: Oh my bad...jeez, you took it out. I read it as an addition.

**Pull request merged**